packages/components/pie-notification/src/index.ts

```diff
@@ -1,24 +1,241 @@
-import { LitElement, html, unsafeCSS } from 'lit';
-
-import { defineCustomElement } from '@justeattakeaway/pie-webc-core';
+import {
+    unsafeCSS,
+    nothing,
+    LitElement,
+    type TemplateResult,
+    type PropertyValues,
+} from 'lit';
+import { type StaticValue, html, unsafeStatic } from 'lit/static-html.js';
+import { defineCustomElement, validPropertyValues } from '@justeattakeaway/pie-webc-core';
+import { property, queryAssignedElements, state } from 'lit/decorators.js';
+import { type NotificationProps, variants, headingLevels } from './defs';
 import styles from './notification.scss?inline';
-import { NotificationProps } from './defs';
+
+import '@justeattakeaway/pie-icon-button';
+import '@justeattakeaway/pie-icons-webc/IconClose';
+import '@justeattakeaway/pie-icons-webc/IconInfoCircle';
+import '@justeattakeaway/pie-icons-webc/IconAlertCircle';
+import '@justeattakeaway/pie-icons-webc/IconAlertTriangle';
+import '@justeattakeaway/pie-icons-webc/IconCheckCircle';
 
 // Valid values available to consumers
 export * from './defs';
 
 const componentSelector = 'pie-notification';
+const componentClass = 'c-notification';
+
+type InternalVariantType = Exclude<NotificationProps['variant'], undefined>;
+type InternalHeadingLevelType = Exclude<NotificationProps['headingLevel'], undefined>;
 
 /**
  * @tagname pie-notification
  */
 export class PieNotification extends LitElement implements NotificationProps {
-    render () {
-        return html`<h1 data-test-id="pie-notification">Hello world!</h1>`;
-    }
+    @property({ type: Boolean })
+    public isOpen = true;
+
+    @property()
+    @validPropertyValues(componentSelector, variants, 'neutral')
+    public variant: InternalVariantType = 'neutral';
+
+    @property({ type: Boolean })
+    public isCompact = false;
+
+    @property({ type: String })
+    public heading!: string;
+
+    @property()
+    @validPropertyValues(componentSelector, headingLevels, 'h2')
+    public headingLevel: InternalHeadingLevelType = 'h2';
+
+    @property({ type: Boolean })
+    public hideIcon = false;
+
+    @property({ type: Boolean })
+    public hideCloseIcon = false;
+
+    @queryAssignedElements({ slot: 'icon' }) _iconSlot!: Array<HTMLElement>;
```

Comment:
Nit: Slightly easier to read and consistent with other declarations
```suggestion
    @queryAssignedElements({ slot: 'icon' })
    _iconSlot!: Array<HTMLElement>;
```

Reply:
back in the day my IDE was complaining about this formatting but now it is not complaining anymore 😅

Reply:
Correction, I had to restart the lint server so I got the error:
![Screenshot 2024-02-20 at 11 30 41 AM](https://github.com/justeattakeaway/pie/assets/7768202/7424c2aa-17bf-42f4-9150-906b9a6342a4)
This is how we make the linter happy which I prefer the solution already implemented instead this one:
![Screenshot 2024-02-20 at 11 31 17 AM](https://github.com/justeattakeaway/pie/assets/7768202/b52d2dd2-e3f5-4aa0-b2d9-00589ea63435)